From ea140035d01afe28b896b6cb9848f2b2824d1bd2 Mon Sep 17 00:00:00 2001
Message-Id: <ea140035d01afe28b896b6cb9848f2b2824d1bd2.1659423146.git.lin.liu@citrix.com>
From: Jan Beulich <jbeulich@suse.com>
Date: Fri, 18 Feb 2022 14:46:27 +0100
Subject: video/vesa: adjust (not just) command line option handling

Document the remaining option. Add section annotation to the variable
holding the parsed value as well as a few adjacent ones. Adjust the
types of font_height and vga_compat.

Signed-off-by: Jan Beulich <jbeulich@suse.com>
Acked-by: Andrew Cooper <andrew.cooper3@citrix.com>

diff --git a/docs/misc/xen-command-line.pandoc b/docs/misc/xen-command-line.pandoc
index d18b6fdcec44..5b11240f6671 100644
--- a/docs/misc/xen-command-line.pandoc
+++ b/docs/misc/xen-command-line.pandoc
@@ -2318,6 +2318,11 @@ cache-warming. 1ms (1000) has been measured as a good value.
 ### vesa-ram
 > `= <integer>`
 
+> Default: `0`
+
+This allows to override the amount of video RAM, in MiB, determined to be
+present.
+
 ### vga
 > `= ( ask | current | text-80x<rows> | gfx-<width>x<height>x<depth> | mode-<mode> )[,keep]`
 
diff --git a/xen/drivers/video/vesa.c b/xen/drivers/video/vesa.c
index 551d15568983..92b75902ec9c 100644
--- a/xen/drivers/video/vesa.c
+++ b/xen/drivers/video/vesa.c
@@ -18,16 +18,16 @@
 
 static void lfb_flush(void);
 
-static unsigned char *lfb;
-static const struct font_desc *font;
-static bool_t vga_compat;
+static unsigned char *__read_mostly lfb;
+static const struct font_desc *__initdata font;
+static bool __initdata vga_compat;
 
-static unsigned int vram_total;
+static unsigned int __initdata vram_total;
 integer_param("vesa-ram", vram_total);
 
 static unsigned int __initdata vram_remap;
 
-static int font_height;
+static unsigned int __initdata font_height;
 static int __init parse_font_height(const char *s)
 {
     if ( simple_strtoul(s, &s, 10) == 8 && (*s++ == 'x') )
